test/compile_errors.zig

```diff
@@ -389,6 +389,110 @@ pub fn addCases(cases: *tests.CompileErrorContext) void {
         "tmp.zig:5:29: error: invalid token: '.'",
     });
 
+    cases.add("invalid underscore placement in float literal - 1",
+        \\fn main() void {
+        \\    var bad: f128 = 0._0;
+        \\})
```

Comment:
- please remove invalid trailing token `)` from new tests
- add some compiler-error tests for integer literals ending in underscore:
    - `1000_`
    - `0x1000_`
    - `0b1111_`
    - `0o7777_`